### PR TITLE
corepack: new Portfile

### PR DESCRIPTION
**Pull request by Robotex**
#### Description

I've written this port to add corepack support back to nodejs18

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.6 21G115 arm64
Xcode 14.0.1 14A400

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: devel/corepack/Portfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+name                corepack
+version             0.15.0
+revision            0
+
+categories          devel
+license             MIT
+maintainers         @Robotex openmaintainer
+
+description         Zero-runtime-dependency package acting as bridge between Node projects and their package managers
+long_description    Corepack is a zero-runtime-dependency Node.js script that acts as a bridge between Node.js projects and the package managers they are intended to be used with during development
```
Comment (catap) [inline devel/corepack/Portfile:15]: maybe truncate line to make it easy to read?

Comment (Robotex) [reply devel/corepack/Portfile]: I haven't seen anywhere in the documentation what's the convention for Portfile line lengths, so I just truncated it to keep it below 120 characters

Comment (catap) [reply devel/corepack/Portfile]: Thanks! It seems amazing!

**Comment by catap** [comment]
BTW you're state that you have referenced existing tickets on Trac with full URL in your checklist. But I haven't found it :(

**Comment by Robotex** [comment]
> BTW you're state that you have referenced existing tickets on Trac with full URL in your checklist. But I haven't found it :(

It's in the commit message but for better visibility [here it is](https://trac.macports.org/ticket/65551)

**Review by catap — CHANGES_REQUESTED** [review]
This port can't be installed when `yarn` or `pnpm` is installed.

It falis like this:
```
Error: Failed to activate pnpm: Image error: /opt/local/bin/pnpm is being used by the active corepack port.  Please deactivate this port first, or use 'port -f activate pnpm' to force the activation.
```
and
```
Error: Failed to activate yarn: Image error: /opt/local/bin/yarn is being used by the active corepack port.  Please deactivate this port first, or use 'port -f activate yarn' to force the activation.
```

**Comment by Robotex** [comment]
> This port can't be installed when `yarn` or `pnpm` is installed.
> 
> It falis like this:
> 
> ```
> Error: Failed to activate pnpm: Image error: /opt/local/bin/pnpm is being used by the active corepack port.  Please deactivate this port first, or use 'port -f activate pnpm' to force the activation.
> ```
> 
> and
> 
> ```
> Error: Failed to activate yarn: Image error: /opt/local/bin/yarn is being used by the active corepack port.  Please deactivate this port first, or use 'port -f activate yarn' to force the activation.
> ```

The `conflicts           pnpm yarn` declaration should already halt the installation if yarn or pnpm ports are already installed:
```
robotex@MacBook-Pro-di-Robotex ~ % sudo port install corepack
--->  Computing dependencies for corepack
Error: Can't install corepack because conflicting ports are active: pnpm
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port corepack failed
```
However I'm not sure about pnpm and yarn ports when installed _after_ corepack was installed, should I declare the conflict with this port on their Portfiles in this same PR?

**Comment by catap** [comment]
@Robotex I think yes, you should and this can be one of possible solution.

Another one is make your new port is depend on `yarn` and `pnpm`, and remove that it had installed.

Personally I suggest to the second way because it much cleaner.

**Comment by Robotex** [comment]
After careful analysis of how nodejs gets and packages its corepack integration ([here](https://github.com/nodejs/node/blob/main/Makefile#L1145) and [here](https://github.com/nodejs/node/blob/main/tools/install.py#L115)), I'm wondering if  maybe would be a better choice (than the default strategy) to keep the shims optional so that this port installs only the corepack binary just like the default nodejs install does and leave to the user the responsability to enable the wanted shims.
Or the opposite way, install all shims by default and declare conflict with all the three package managers.

> Another one is make your new port is depend on `yarn` and `pnpm`, and remove that it had installed.
> 
> Personally I suggest to the second way because it much cleaner.

@catap Would you mind explaining your second route? 

> (and sorry for nitpicking)

No worries, I'm open minded to better solutions that work for everyone :)

File: devel/corepack/Portfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           npm 1.0
+
+name                corepack
+version             0.15.0
```
Comment (reneeotten) [inline devel/corepack/Portfile:7]: please update to latest upstream version - currently 0.19.0

File: devel/corepack/Portfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+revision            0
+
+categories          devel
+license             MIT
+maintainers         @Robotex openmaintainer
```
Comment (reneeotten) [inline devel/corepack/Portfile:12]: please add your obfuscate e-mail address as well ```{@jverne example.org:julesverne}```

**Review by reneeotten — CHANGES_REQUESTED** [review]
@Robotex going through our PR queue in an attempt to get old(er) PRs move towards completion. Can you please take care of the two requests and I'll merge.

**Comment by catap** [comment]
@reneeotten am I mistaken or `yarn` and `pnpm` ports should be also updated by adding `conflicts`?

**Comment by reneeotten** [comment]
No response in 3 months... closing this PR, feel free to reopen once you found the time to finalize it.

**Pull request closed without merging**